### DISCUSSION TITLE
Choppy stuttering every few seconds?

### DISCUSSION THREAD

**Discussion by T3HM3T4L** [smplayer-dev/smplayer#66] [Q&A]
I've played around with a bunch of the settings and nothing seemed to help.
I really like the program but I can't seem to get rid of some slight stuttering that happens over and over again.

It tends to be most notable while playing anime .mkv files, whenever the camera is slowly panning.
I've tried the frame drop settings, hardware decoding, loop filter but nothing seems to work.

I'll gladly switch to using Smplayer forever (as I'm moving to Linux from Windows soon) but this problem is very annoying.

**Comment by dummifiedme** [comment]
Same here. Any solution that you can share?
I didn't have any problems for a long time but for the last few months, I have been facing this issue of frame drops. There is a lot of stuttering and also sometimes the audio plays while the video progresses really really slowly (many frames are dropped I suppose).

**Reply by T3HM3T4L** [reply]
Hi there.
I don't remember if I ever found a solution.
It might've also have something to do with having to force "full composition pipeline" in the X11 settings on my Linux distribution, but I can't say for certain.

Afterwards I started using Celluloid instead, now regular MPV since I figured out how to configure it.

**Comment by smplayer-dev** [comment]
Do you have the problem if you select another video output? (Preferences -> General -> Video)

**Comment by dummifiedme** [comment]
I had 'other' since a long time which was eventually set by default to MPV
folder. Today I changed it to MPV specifically in the drop down. I don't
see any difference. And I changed the driver setting to gpu today.

Is there a recommended setting which i should use?

I think it has problem with high resolution videos (1080p).

**Comment by smplayer-dev** [comment]
Try increasing the threads for decoding and / or enabling hardware decoding in preferences -> performance.

**Comment by ghost** [comment]
Sorry for barging in. Having the same - or very similar - situation: image stuttering every second while sound runs normally. No other applications running. Most notably on HEVC files but not only.

Most likely some update introduced this bug, but no idea if it was a newer **mpv** version, SMPlayer itself or a combination of the two.
Some time ago tried to run with **mplayer** as engine but didn't work well so can't change the engine.

Hardware decoding has always been on Auto, Threads were 2, now increased to 4. Tried with Direct Rendering off and on. Nothing helps.
Sometimes when stuttering is very "heavy" a Cinnamon restart reduces the amplitude (length) of the microfreezes but does not eliminate them.

System is Linux Mint 19.2 x64 Cinnamon on a dual-core i5 M460 CPU with 4GB RAM and NVIDIA GT218M (GeForce 310M) running proprietary driver 340.108.

**Comment by PaPa1ya** [comment]
I had a similar problem with nvidia, for some reason the default setting in 'Performance > Hardware decoding' was not 'auto' but 'none'.

**Comment by anubisltx** [comment]
Ubuntu 23.10 with NVIDIA 3090 - I set my threads to maximum and my 'hardware decoding' to cuda and it seemed to help a lot - vdpau for some reason did not.

**Reply by Raza-0007** [reply]
For the latest Nvidia GPU's like the 3090, the recommended mpv hwdec is nvdec. Both cuda and vdpau have been depreciated and not actively developed. See here:

https://mpv.io/manual/master/#options-hwdec

You can enable nvdec by keeping hwdec option to auto in smplayer and then passing this line in `Options> Preferences> Advanced> Mplayer/mpv> options`

`-hwdec=nvdec`

**Reply by anubisltx** [reply]
Legend - thank you very much for that! I am very out of touch these days.

**Reply by Raza-0007** [reply]
You are welcome. 

I do believe that smplayer's hwdec code needs to be modified to include only the mpv recommended hwdecs. On my system, Windows 10, smplayer gives me the options of `d3d11va`, `dxva2`, `cuda` and `crystalhd`. Out of these only `d3d11va` is on mpv's recommended list. 

However, even on Windows 10, if the GPU is Nvidia, then the best hwdec performance will be obtained by using `nvdec`, which is a successor to `cuda`.

**Reply by octimis** [reply]
For an nVidia Geforce 700 Series (7xx) graphics card from 2013 designated 'GK104' by 'sudo lshw -c video' and x11 protocol instead of wayland. What would be recommended setting?

I'm experriencing 2 second delays that don't exist on MPV standalone when seeking on 4k videos, and sometimes (1 in 20 chance) SMPlayer will just flat out freeze, I.e. can't resume video playback.

**Reply by Raza-0007** [reply]
@octimis, if standalone mpv is working without issues then print out its log and see which options it is choosing by default. Then compare this with the mpv log when mpv is running in smplayer. If the options are different then change them from within smplayer.

I have limited experience with linux and no experience using smplayer on linux, so I cannot recommend optimal settings for you. But try the following settings, they may be of some help.

First restore smplayer to default settings, and then just change the following:

profile=fast (on mpv 0.37 or newer)
vo=gpu-next or gpu
hwdec=nvdec
priority=abovenormal

~Raza

**Comment by Mrnofish** [comment]
For anybody still experiencing this issue, try this from a shell:

`pgrep smplayer | sudo xargs renice -15 -p`

It will increase the priority of the process, and IME it reduces or eliminates the stuttering depending on the general pressure on the system. For a belt and suspender approach also apply a second line that targets specifically mpv (replace smplayer with mpv)

This should be more useful if the video card has poor or no hardware acceleration for the codec the video has been encoded with.

I.e. it shouldn't be necessary with h.264 in 99.9% of the instances.

The only downside I've found so far, besides the inconvenience of having to run the script manually, is that renice needs elevation. There are probably workarounds, though.
